server/game/cards/events/01/tearsoflys.js

```diff
@@ -5,32 +5,22 @@ class TearsOfLys extends DrawCard {
         this.reaction({
             max: ability.limit.perChallenge(1),
             when: {
-                afterChallenge: ({challenge}) => (
-                    challenge.attackingPlayer === this.controller &&
-                    challenge.winner === this.controller &&
-                    challenge.challengeType === 'intrigue'
-                )
+                afterChallenge: event => event.challenge.attackingPlayer === this.controller && event.challenge.winner === this.controller &&
+                                         event.challenge.challengeType === 'intrigue'
             },
-            handler: () => {
```

Comment:
Seems like this choice was made for a reason, don't think we should be undoing that